pwndbg/dbg/lldb.py

```diff
@@ -9,22 +13,363 @@
+
+        return LLDBValue(deref)
+
+    @override
+    def string(self) -> str:
+        addr = self.inner.unsigned
+        error = lldb.SBError()
+
+        # Read strings up to 4GB.
+        last_str = None
+        buf = 256
+        for i in range(8, 33):
+            s = self.inner.process.ReadCStringFromMemory(addr, buf, error)
+            if error.Fail():
+                raise pwndbg.dbg_mod.Error(f"could not read value as string: {error.description}")
+            if last_str is not None and len(s) == len(last_str):
+                break
+            last_str = s
+
+            buf *= 2
+
+        return last_str
+
+    @override
+    def fetch_lazy(self) -> None:
+        # Not needed under LLDB.
+        pass
+
+    @override
+    def __int__(self) -> int:
+        return self.inner.signed
+
+    @override
+    def cast(self, type: pwndbg.dbg_mod.Type | Any) -> pwndbg.dbg_mod.Value:
+        assert isinstance(type, LLDBType)
+        t: LLDBType = type
+
+        return LLDBValue(self.inner.cast(t.inner))
+
+        return LLDBValue(self.inner.Cast(type.inner))
+
+
+class LLDBProcess(pwndbg.dbg_mod.Process):
+    def __init__(self, process: lldb.SBProcess, target: lldb.SBTarget):
+        self.process = process
+        self.target = target
+
+    @override
+    def evaluate_expression(self, expression: str) -> pwndbg.dbg_mod.Value:
+        value = self.target.EvaluateExpression(expression)
+        opt_out = _is_optimized_out(value)
+
+        if not value.error.Success() and not opt_out:
+            raise pwndbg.dbg_mod.Error(value.error.description)
+
+        return LLDBValue(value)
+
+
+class LLDBCommand(pwndbg.dbg_mod.CommandHandle):
+    def __init__(self, handler_name: str, command_name: str):
+        self.handler_name = handler_name
+        self.command_name = command_name
+
+
 class LLDB(pwndbg.dbg_mod.Debugger):
+    exec_states: List[lldb.SBExecutionState]
+
     @override
     def setup(self, *args):
+        self.exec_states = []
+
         debugger = args[0]
         assert (
             debugger.__class__ is lldb.SBDebugger
         ), "lldbinit.py should call setup() with an lldb.SBDebugger object"
 
+        module = args[1]
+        assert module.__class__ is str, "lldbinit.py should call setup() with __name__"
+
+        self.module = module
         self.debugger = debugger
 
+        import pwndbg.commands
+
+        pwndbg.commands.load_commands()
+
+        import argparse
+
+        parser = argparse.ArgumentParser(description="Prints a test message.")
+
+        @pwndbg.commands.ArgparsedCommand(parser)
+        def test2():
+            print("Test 2!")
+
+    @override
+    def add_command(
+        self, command_name: str, handler: Callable[[pwndbg.dbg_mod.Debugger, str, bool], None]
+    ) -> pwndbg.dbg_mod.CommandHandle:
+        debugger = self
+
+        class CommandHandler:
+            def __init__(self, debugger, _):
+                pass
```

Comment:
I wonder if this should maybe inherit from some base class or sth.

Reply:
Just to be sure, we define the class here bcoz we need to "register it" in sys.modules?

Reply:
Yep, LLDB takes classes for commands, not class instances, so we have to create a new class for every new command that we want to register. I think it's important to point out that we'd have to create a new class regardless of where we were to put it. 

The fact that we have to add it to the dictionary of globals of `lldbinit` is due to another limitation with LLDB, which needs us to have our classes under globally-accessible names when we add new commands.

Reply:
Pls add a comment mentioning that here